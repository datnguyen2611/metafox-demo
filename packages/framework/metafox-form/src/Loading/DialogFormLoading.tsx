/**
 * @type: ui
 * name: form.DialogLoadingComponent
 */
import React from 'react';
import { CircularProgress, Box } from '@mui/material';

export default function DefaultLoading() {
  return (
    <Box
      sx={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        minHeight: '300px',
        minWidth: '300px'
      }}
    >
      <CircularProgress size={32} />
    </Box>
  );
}
