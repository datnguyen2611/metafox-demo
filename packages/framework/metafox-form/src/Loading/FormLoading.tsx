/**
 * @type: ui
 * name: form.DefaultLoading
 */
import React from 'react';
import { CircularProgress, Box } from '@mui/material';

export default function DefaultLoading() {
  return (
    <Box
      sx={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        p: 2
      }}
    >
      <CircularProgress size={24} />
    </Box>
  );
}
