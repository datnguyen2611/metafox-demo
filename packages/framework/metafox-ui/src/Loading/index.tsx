/**
 * @type: ui
 * name: Loading
 */
import { styled, useTheme } from '@mui/material/styles';
import React from 'react';
import LoadingSVG from './LoadingSVG.svg';
import LoadingSVGDark from './LoadingSVGDark.svg';
import { useGlobal } from '@metafox/framework';

type LoadingVariant = 'init' | 'absolute' | 'fixed' | 'relative' | undefined;

interface LoadingProps {
  related?: boolean;
  center?: boolean;
  absolute?: boolean;
  variant?: LoadingVariant;
}

const LoadingRoot = styled('div', {
  name: 'Loading',
  slot: 'Root',
  shouldForwardProp: prop =>
    prop !== 'related' && prop !== 'darkMode' && prop !== 'center'
})<{
  darkMode?: boolean;
  variant?: LoadingVariant;
  related?: boolean;

  center?: boolean;
  absolute?: boolean;
}>(({ theme, variant, darkMode }) => ({
  ...(variant === 'absolute' && {
    display: 'flex',
    justifyContent: 'center',
    minHeight: '100%',
    position: 'absolute',
    left: '0',
    top: '0',
    right: '0',
    bottom: '0',
    alignItems: 'center',
    backgroundColor: darkMode ? process.env.MFOX_LOADING_BG : '#fff'
  }),
  ...((variant === 'fixed' || !variant) && {
    display: 'flex',
    justifyContent: 'center',
    minHeight: '100%',
    position: 'fixed',
    left: '0',
    top: '0',
    right: '0',
    bottom: '0',
    alignItems: 'center',
    backgroundColor: darkMode ? process.env.MFOX_LOADING_BG : '#fff',
    zIndex: 99,
    opacity: 0.5
  }),
  ...(variant === 'init' && {
    display: 'flex',
    justifyContent: 'center',
    minHeight: '100%',
    position: 'fixed',
    left: '0',
    top: '0',
    right: '0',
    bottom: '0',
    alignItems: 'center',
    backgroundColor: darkMode ? process.env.MFOX_LOADING_BG : '#fff',
    zIndex: 99
  }),
  ...(variant === 'relative' && {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: darkMode ? process.env.MFOX_LOADING_BG : '#fff'
  })
}));

export default function Loading({
  related,
  center,
  absolute,
  variant
}: LoadingProps) {
  const theme = useTheme();
  const { cookieBackend } = useGlobal();
  const themeType = cookieBackend.get('themeType');
  const themeTypeMode = themeType ?? theme.palette.mode;
  const darkMode = themeTypeMode === 'dark';

  if (!variant) {
    if (absolute) {
      variant = 'absolute';
    } else if (related && center) {
      variant = 'relative';
    }
  }

  return (
    <LoadingRoot
      variant={variant}
      darkMode={darkMode}
      data-testid="loadingIndicator"
    >
      <div style={{ width: '5rem', height: '5rem' }}>
        {darkMode ? <LoadingSVGDark /> : <LoadingSVG />}
      </div>
    </LoadingRoot>
  );
}
