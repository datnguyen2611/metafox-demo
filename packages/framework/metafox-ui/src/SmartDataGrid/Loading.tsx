import { CircularProgress } from '@mui/material';
import React from 'react';

const styles: {
  backdrop: React.CSSProperties;
  content: React.CSSProperties;
} = {
  backdrop: {
    position: 'absolute',
    left: 0,
    top: 0,
    bottom: 0,
    right: 0
  },
  content: {
    position: 'absolute',
    left: '50%',
    top: '50%',
    marginLeft: -20,
    marginTop: -20
  }
};

export default function Loading({ size = 40 }: { size?: number }) {
  return (
    <div style={styles.backdrop} data-testid="loadingIndicator">
      <div style={styles.content}>
        <CircularProgress variant="indeterminate" size={size} />
      </div>
    </div>
  );
}
