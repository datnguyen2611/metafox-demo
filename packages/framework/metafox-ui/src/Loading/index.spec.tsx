import * as React from 'react';
import renderer from 'react-test-renderer';
import Loading from './index';

describe('Loading', () => {
  it('<Loading />', async () => {
    const loading = renderer.create(<Loading />).toJSON();
    expect(loading).toMatchSnapshot();
  });

  it('<Loading related/>', async () => {
    const loading = renderer.create(<Loading related />).toJSON();
    expect(loading).toMatchSnapshot();
  });

  it('<Loading center/>', async () => {
    const loading = renderer.create(<Loading center />).toJSON();
    expect(loading).toMatchSnapshot();
  });

  it('<Loading related center/>', async () => {
    const loading = renderer.create(<Loading related center />).toJSON();
    expect(loading).toMatchSnapshot();
  });

  it('<Loading darkMode/>', async () => {
    window.matchMedia = jest.fn().mockReturnValue({ matches: 1 });
    const loading = renderer.create(<Loading />).toJSON();
    expect(loading).toMatchSnapshot();
  });
});
