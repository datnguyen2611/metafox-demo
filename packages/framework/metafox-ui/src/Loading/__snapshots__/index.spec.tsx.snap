// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading <Loading /> 1`] = `
<div
  className="css-1mqnh9d-Loading-root"
  data-testid="loadingIndicator"
>
  <div
    style={
      {
        "height": "5rem",
        "width": "5rem",
      }
    }
  >
    < />
  </div>
</div>
`;

exports[`Loading <Loading center/> 1`] = `
<div
  className="css-1mqnh9d-Loading-root"
  data-testid="loadingIndicator"
>
  <div
    style={
      {
        "height": "5rem",
        "width": "5rem",
      }
    }
  >
    < />
  </div>
</div>
`;

exports[`Loading <Loading darkMode/> 1`] = `
<div
  className="css-1mqnh9d-Loading-root"
  data-testid="loadingIndicator"
>
  <div
    style={
      {
        "height": "5rem",
        "width": "5rem",
      }
    }
  >
    < />
  </div>
</div>
`;

exports[`Loading <Loading related center/> 1`] = `
<div
  className="css-2ndnsb-Loading-root"
  data-testid="loadingIndicator"
  variant="relative"
>
  <div
    style={
      {
        "height": "5rem",
        "width": "5rem",
      }
    }
  >
    < />
  </div>
</div>
`;

exports[`Loading <Loading related/> 1`] = `
<div
  className="css-1mqnh9d-Loading-root"
  data-testid="loadingIndicator"
>
  <div
    style={
      {
        "height": "5rem",
        "width": "5rem",
      }
    }
  >
    < />
  </div>
</div>
`;
